Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/DvachApi.java

```diff
@@ -0,0 +1,238 @@
+                }
+
+                reader.endArray();
+            } else {
+                reader.skipValue();
+            }
+        }
+
+        reader.endObject();
+    }
+
+    @Override
+    public void loadCatalog(JsonReader reader, ChanReaderProcessingQueue queue) throws Exception {
+        reader.beginObject(); // Main object
+
+        while (reader.hasNext()) {
+            if (reader.nextName().equals("threads")) {
+                reader.beginArray(); // Threads array
+
+                while (reader.hasNext()) {
+                    readPostObject(reader, queue);
+                }
+
+                reader.endArray();
+            } else {
+                reader.skipValue();
+            }
+        }
+
+        reader.endObject();
+    }
+
+    @Override
+    public void readPostObject(JsonReader reader, ChanReaderProcessingQueue queue) throws Exception {
+        Post.Builder builder = new Post.Builder();
+        builder.board(queue.getLoadable().board);
+
+        SiteEndpoints endpoints = queue.getLoadable().getSite().endpoints();
+
+        List<PostImage> files = new ArrayList<>();
+
+        reader.beginObject();
+        while (reader.hasNext()) {
+            String key = reader.nextName();
+
+            switch (key) {
+                case "no":
+                    builder.id(Integer.parseInt(reader.nextString()));
+                    break;
+                case "name":
+                    builder.name(reader.nextString());
+                    break;
+                case "comment":
+                    builder.comment(reader.nextString());
+                    break;
+                case "timestamp":
+                    builder.setUnixTimestampSeconds(reader.nextLong());
+                    break;
+                case "trip":
+                    builder.tripcode(reader.nextString());
+                    break;
+                case "op":
+                    int opId = reader.nextInt();
+                    builder.op(opId == 0);
+                    builder.opId(opId);
+                    break;
+                case "sticky":
+                    builder.sticky(reader.nextInt() == 1);
+                    break;
+                case "closed":
+                    builder.closed(reader.nextInt() == 1);
+                    break;
+                case "archived":
+                    builder.archived(reader.nextInt() == 1);
+                    break;
+                case "posts_count":
+                    builder.replies(reader.nextInt() - 1);
+                    break;
+                case "files_count":
+                    builder.images(reader.nextInt());
+                    break;
+                case "lasthit":
+                    builder.lastModified(reader.nextLong());
+                    break;
+                case "num":
+                    String num = reader.nextString();
+                    builder.id(Integer.parseInt(num));
+                    builder.posterId(num);
```

Comment:
`posterId` is used in 4chan for the unique id assigned to unique ip addresses.
The way it is used now causes all posts to have "ID: nnn" on them.